ambry-store/src/main/java/com.github.ambry.store/PersistentIndex.java

```diff
@@ -496,35 +500,40 @@ private boolean needToRollOverIndex(IndexEntry entry) {
    * @throws StoreException
    */
   IndexValue findKey(StoreKey key) throws StoreException {
-    return findKey(key, null);
+    return findKey(key, null, EnumSet.of(IndexEntryType.PUT, IndexEntryType.DELETE));
```

Comment:
Shouldn't the third argument be `EnumSet.allOf(IndexEntryType.class)`?

Reply:
No - the normal `findKey` will never return a TTL update record because none of its users care for it (everyone wants either a put or delete) - I also thing this is semantically true (i.e. not just the case today) because only specialized callers will need things like update records. Everyone else will want a put or delete

Reply:
In fact you could argue that everyone is only interested in the PUT entry with the flags updated to reflect current state